app/src/main/java/com/example/lastfm/MainActivity.java

```diff
@@ -39,6 +41,13 @@ public void onRefresh() {
 
     private void retrofitRequest() {
         LastFMClient client = ServiceGenerator.getInstance().getLastFMClient();
+        final ProgressDialog progressDoalog;
+        progressDoalog = new ProgressDialog(MainActivity.this);
```

Comment:
u must add a progress bar to the layout of the activity. Also, don't forget about ArtistInfoActivity

Reply:
Im using progress dialog not progress bar, progress dialog dont have layout parameters,than should i use progress bar instead of progress dialog?

Reply:
Changed progressdialog to progressbar